feat: Enable Antrea plugin on Windows Nodes

reachjainrahul opened the pull request:
**Reason for Change**:

Antrea can be enabled as networkPlugin and networkPolicy plugin
on Windows node. Antrea windows plugin has dependency on Openvswitch.
It requires signed version of upstream OVS. Here, we install OVS
with test signing mode enabled on Windows node. Its not recommended
for production.

Also updating antrea version to v0.9.3

**Issue Fixed**:

**Requirements**:

- [X] uses [conventional commit messages](https://www.conventionalcommits.org/)
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

reachjainrahul commented:
@jackfrancis could you please help assign this review to right person?

jsturtevant commented on parts/k8s/addons/antrea-windows.yaml:
```diff
@@ -0,0 +1,189 @@
+    # wins will rename the binary when executing it. So we need to copy the binary everytime before running it.
+    mkdir -force /host/k/antrea/bin
+    cp /k/antrea/bin/* /host/k/antrea/bin/
+    C:/k/antrea/utils/wins.exe cli process run --path /k/antrea/bin/antrea-agent.exe --args "--config=/k/antrea/etc/antrea-agent.conf --logtostderr=false --log_dir=/k/antrea/logs/ --alsologtostderr --log_file_max_size=100 --log_file_max_num=4" --envs "KUBERNETES_SERVICE_HOST=$env:KUBERNETES_SERVICE_HOST KUBERNETES_SERVICE_PORT=$env:KUBERNETES_SERVICE_PORT ANTREA_SERVICE_HOST=$env:ANTREA_SERVICE_H ...
```
Can we get this configured, installed and running without wins?  We don't use wins to configure the rest of the components and currently don't want to introduce the component.

ruicao93 replied:
Hi @jsturtevant , Antrea depends on wins to manage antrea-agent and kube-process processes from Pod.
The way is also the official recommendation: https://kubernetes.io/docs/tasks/administer-cluster/kubeadm/adding-windows-nodes/

Are you just want avoid to introduce wins or any this kind of tools? Without wins like tools we will must start antrea-agent process directly from host and do lots of configuration for it.

jsturtevant replied:
We don't want to introduce wins at this point for aks-engine.  Wins is a different way of configuring a cluster and aks-engine comes from a time before that was a way of doing things.  Introducing a second way of configuring things is out of scope for aks-engine at this point.

ruicao93 replied:
OK, get it. We will add scripts help to start antre-agent without wins.

jsturtevant replied:
thanks

jsturtevant replied:
There currently isn't a way to start a process on the host from a POD.  Services running on the host still have access to the k8s control plane, this is the way kubelet and the other cni's are running currently.  

There is a draft KEP related to enabling privileged containers to solve this problem outside of k8s. Please add some comments to make sure it fits your needs: https://docs.google.com/document/d/12EUtMdWFxhTCfFrqhlBGWV70MkZZPOgxw0X-LTR0VAo/edit#heading=h.kpsimyb2q1sg

reachjainrahul replied:
@jsturtevant We have added antrea as a windows service. Could you please take a look at the updated review. Hopefully we can checkin soon.

I ran E2E tests on Windows and it passed with Antrea v0.9.0

jsturtevant commented on parts/k8s/windowsantreacnifunc.ps1:
```diff
@@ -0,0 +1,42 @@
+    & Bcdedit.exe -set TESTSIGNING ON
+
+    # TODO: Discuss with AKS Folks, how to install Upstream OVS on Windows Node
+    # and sign it.
```
Can you explain more on the requirement for signing the ovs? For aks-engine users will they be able use this as is or is there another process that is required for them to get signed binaries?

wenyingd replied:
Antrea is using the Open Source OVS as the forwarding component, nd the Open Source OVS has no certificate from Microsoft. Hence, we have to enable the "testsigning" mode on the Windows host to ensure OVS datapath could forward packets correctly. 

But there might be some security risks to enable the "testsigning" mode on a Windows host, so we want to get more thoughts from the AKS folks to see if we can we get signed for the Open Source OVS (then we don't need to enable the testsigning mode), or the risk for the "testsigning" mode is acceptable (then we could continue using the unsigned Open Source OVS).

jsturtevant replied:
I don't have any experience with OVS so have a few more questions: 

- For folks who don't use AKS and only use aks-engine but would like to use OVS how will they get the signed cert? 
- Do they need to rely on Microsoft to provide a certificate?  Will they want to provide a cert that they sign?  
- In the case of AKS, will customers also want to provide there own certs?

Want to make sure there is a path for teams and users that don't use AKS as well.

@AbelHu might have some additional questions or insight as well.

wenyingd replied:
Without a Microsoft official certificate, "testsigning" mode must be enabled even if the user generates a self-signed cert.

To acquire an official certificate, we need an organization. However, either the open source OVS community or the Linux Foundation doesn't provide the signed cert currently. So the OVS community recommends users to enable "testsigning" mode when using OVS on Windows host.

An official certificate requires all the binaries in OVS should be signed that also includes the installers, catalog files and driver files. The action owner should be an orgnization first, and has an account in Microsoft to trigger the signing process. A series of tests playlist that will run against some nodes which already have OVS deployed. Once all the tests have been passed, a signed package can be created and uploaded to Microsoft for verification and further AV scanning.

jsturtevant replied:
so the binaries need to be signed and there isn't a cert that needs to be provided? This means Microsoft could sign the binaries out of band and distribute them many customers?  We also require our binaries and powershell scripts to be signed (#3441 for example)

I had initially thought there was a requirement for a private cert that needed to be installed on the node per customer.

reachjainrahul replied:
Yes @jsturtevant. If Microsoft can sign the binary and put in under Azure storage, we can update our install script to fetch the binary from that location. 
Also, it is possible to consider a new Windows VHD from Microsoft with OVS already installed on it?

jsturtevant replied:
Thanks this makes sense. We do something similar for other packages.  Given this isn't GA yet we can leave the self signing for folks to try it out. As it matures and demand grows we can address signing.  We will want to make sure that the components are not installed unless the cni is installed this way it doesn't get flagged for signing by any processes that might be checking.

wenyingd replied:
Hi @jsturtevant , the OVS is signed by Microsoft means WHQL certification for OVS binaries.

Following arr the steps to get the WHQL certification:

1. You need to pass the tests for HLK on Windows 10 family and above
(https://docs.microsoft.com/en-us/windows-hardware/test/hlk/) and
HCK on Windows 8 family
(https://docs.microsoft.com/en-us/windows-hardware/drivers/develop/run-the-hck-test-suites-in-the-wdk).
The above is a series of tests playlist that will be ran against some nodes which already have OVS deployed.
The tests are a bit finicky (to say at best) and vary between versions and types (Server / Client – Home, Professional
and Enterprise) of Windows OS.

2. Once you pass all the tests you create a signed package which you upload to Microsoft for verification and
further AV scanning.

To note: The driver certificate that you use on the deployed nodes prior to HCK/HLK testing, has to be signed
using a different, more expensive certificate: EV certificate
(https://docs.microsoft.com/en-us/windows-hardware/drivers/dashboard/get-a-code-signing-certificate)
I’m not sure if they dropped this requirement but it was a huge issue when people wanted to sign drivers
for 8.1/2012 R2 and 10-RS4/ Server 2016.

Again an organization has to be behind that certificate.

WHQL signing means that Microsoft will cross sign the binaries that we pushed in our publisher package.
This will also allow the use of Secure Boot.

jsturtevant replied:
When we address the signing aspect as discussed above we will also need include signing the `AntreaStartWrapper` and `https://github.com/vmware-tanzu/antrea/releases/download/v0.9.0/Start.ps1`.

marosset replied:
I'm not sure if the aks-engine team can perform the type of signing needed here.
We currently perform authenicode signing for many open source components (mainly kubernetes binaries) but that is intended to provide assurance that these binaries were built by Microsoft and not modified.
We don't currently have the capabilities to sign with certificates that allow code / driver execution.

marosset replied:
Also, enabling test signing on AKS will not be acceptable.
We'll need to figure out how to get OVS to be signed properly.

reachjainrahul commented:
@jackfrancis @jsturtevant Could you please help review the updated patch?

jsturtevant commented on parts/k8s/windowsantreacnifunc.ps1:
```diff
@@ -0,0 +1,81 @@
+    )
+
+    Write-Log "Downloading Antrea Start Powershell script"
+    $StartPs = [Io.path]::Combine($KubeDir, "Start.ps1")
```
Can this be named AntreaStart on disk?

reachjainrahul replied:
Renamed

jsturtevant commented on pkg/api/vlabs/validate.go:
```diff
@@ -1828,6 +1826,10 @@ func (a *Properties) validateContainerRuntime(isUpdate bool) error {
 		}
+
+		if a.OrchestratorProfile.KubernetesConfig.NetworkPlugin == NetworkPluginAntrea {
+			return errors.Errorf("networkPlugin antrea for windows is not supported with ContainerRuntime=containerd")
```
Is support for containerd in the roadmap?

reachjainrahul replied:
We havent planned for containerd support yet.

jsturtevant commented on pkg/api/vlabs/validate_test.go:
```diff
@@ -934,13 +934,6 @@ func Test_Properties_ValidateNetworkPolicy(t *testing.T) {
 	}
-
-	p.OrchestratorProfile.KubernetesConfig.NetworkPolicy = NetworkPolicyAntrea
-	if err := p.OrchestratorProfile.KubernetesConfig.validateNetworkPolicy(k8sVersion, true); err == nil {
```
looking a few lines above it seems the k8sversion passed in is 1.7.9.  This should stay the same and check for 1.18+ should be added.

reachjainrahul replied:
Done.. Updated the patch

reachjainrahul replied:
@jsturtevant I explicitly added for 1.17. Do you still feel the need for 1.18 as 1.17 will cover validation failure

jsturtevant commented:
/azp run

reachjainrahul commented:
/azp run

jsturtevant commented:
/azp run pr-e2e

reachjainrahul commented:
@jsturtevant can we merge this today?

jsturtevant commented on pkg/api/vlabs/validate_test.go:
```diff
@@ -974,13 +974,6 @@ func Test_Properties_ValidateNetworkPlugin(t *testing.T) {
-		t.Errorf(
-			"should error on antrea for windows clusters",
-		)
-	}
```
Can you add a test for the correct versioning of k8s needed?

jsturtevant commented on pkg/api/vlabs/validate.go:
```diff
@@ -1722,9 +1717,12 @@ func (k *KubernetesConfig) validateNetworkPolicy(k8sVersion string, hasWindows b
 	}
 
+	if networkPolicy == NetworkPolicyAntrea && hasWindows && !common.IsKubernetesVersionGe(k8sVersion, "1.18.0") {
+		return errors.New("networkPolicy antrea for windows requires kubernetes version of 1.18 or higher")
```
Can you add a test for this?

marosset replied:
+1

jsturtevant commented:
Need to have tests pass and  two additional items:

- We should add support to the [e2e](https://github.com/Azure/aks-engine/blob/master/test/e2e/test_cluster_configs/network_policy/antrea_azure.json) 
- docs

/azp run pr-e2e

jackfrancis commented:
/azp run pr-e2e

jackfrancis commented:
We want to add a Windows pool to the antrea cluster configs we're testing, these are the relative file paths:

- test/e2e/test_cluster_configs/network_policy/antrea.json
- test/e2e/test_cluster_configs/network_policy/antrea_azure.json

reachjainrahul commented:
> Need to have tests pass and two additional items:
> 
> * We should add support to the [e2e](https://github.com/Azure/aks-engine/blob/master/test/e2e/test_cluster_configs/network_policy/antrea_azure.json)
> * docs
> 
> /azp run pr-e2e

Added new file in e2e for antrea windows. 

I will take care of doc in next checkin (PR).

reachjainrahul commented:
Updated the PR with doc too.. @jackfrancis

jsturtevant commented on examples/networkpolicy/README.md:
````diff
@@ -88,6 +88,9 @@ The kubernetes-antrea deployment template enables Antrea networking and policies
       }
 ```
 
+Antrea can be enabled for both Linux and Windows nodes. For Windows nodes, minimum kubernetes version required is 1.18. Also, Antrea requires Open Source OVS for networking and enforcing network policies. Open Source OVS is not signed. So, TestSigning mode is enabled for Windows node to ensure OVS datapath could forward packets correctly. Currently Antrea for Windows nodes support is in Beta.
````
```suggestion
In Kubernetes 1.18+,  Antrea can be enabled Windows nodes.  Antrea for Windows requires Open Source OVS for networking and enforcing network policies. For the initial implementation in aks-engine, Open Source OVS is not signed. See [this conversation](https://github.com/Azure/aks-engine/pull/3597/files#r458363545) to learn more about the requirements.  TestSigning mode is enabled for Windows node to ensure OVS datapath can forward packets correctly.   Antrea support for Windows nodes is in Beta.
```

reachjainrahul replied:
Done

jsturtevant commented:
/azp run pr-e2e

reachjainrahul commented:
/assign @jsturtevant

jsturtevant commented:
/azp run pr-e2e

reachjainrahul commented:
CodeCov passed with earlier changes. Only thing extra here is the documentation.
Also, I dont think Antrea is tested in E2E. Please advice next steps. @jsturtevant

jackfrancis commented:
Hi @reachjainrahul, I added a Windows pool to the antrea config we already test, and the Windows node is not coming online.

See the changes I made to `test/e2e/test_cluster_configs/network_policy/antrea.json`. That cluster config should work with these changes, right?

reachjainrahul commented:
> Hi @reachjainrahul, I added a Windows pool to the antrea config we already test, and the Windows node is not coming online.
> 
> See the changes I made to `test/e2e/test_cluster_configs/network_policy/antrea.json`. That cluster config should work with these changes, right?

I added antrea_windows.json, because we wanted to test Windows with 1.18 version only. Antrea for Linux can work from 1.15 onwards.. I see you set version to latest. I am OK with the change. Wondering, do you run E2E tests against fixed version, or its round robin, and one of the version from 1.15 to 1.18 is selected??

reachjainrahul commented:
> > Hi @reachjainrahul, I added a Windows pool to the antrea config we already test, and the Windows node is not coming online.
> 
> > 
> 
> > See the changes I made to `test/e2e/test_cluster_configs/network_policy/antrea.json`. That cluster config should work with these changes, right?
> 
> 
> 
> I added antrea_windows.json, because we wanted to test Windows with 1.18 version only. Antrea for Linux can work from 1.15 onwards.. I see you set version to latest. I am OK with the change. Wondering, do you run E2E tests against fixed version, or its round robin, and one of the version from 1.15 to 1.18 is selected??
> 
> 
> 
> 

I hope with ur latest change cluster is coming up. If so, can we merge this today?? @jackfrancis

jackfrancis commented:
@reachjainrahul the test was run against a 1.19 cluster.

Here are some docs that can show you how you can run E2E tests locally, assuming you have an Azure subscription with quota to build a cluster.

reachjainrahul commented:
> @reachjainrahul the test was run against a 1.19 cluster.
> 
> 
> 
> Here are some docs that can show you how you can run E2E tests locally, assuming you have an Azure subscription with quota to build a cluster.

Sorry I missed your point. Is the cluster not coming with up 1.19? I verified on 1.18 and it worked for me earlier

reachjainrahul commented:
> > @reachjainrahul the test was run against a 1.19 cluster.
> > 
> > 
> > 
> > Here are some docs that can show you how you can run E2E tests locally, assuming you have an Azure subscription with quota to build a cluster.
> 
> Sorry I missed your point. Is the cluster not coming with up 1.19? I verified on 1.18 and it worked for me earlier 

Another thing to note, this change includes bootstrap script for windows and it needs to be signed before it can be installed. When I bring up the cluster with 1.18, I point that to my storage location so that antrea bootstrap script gets called. Not sure if you facing any problem with 1.19 cluster due to this.

jackfrancis commented:
> Another thing to note, this change includes bootstrap script for windows and it needs to be signed before it can be installed. When I bring up the cluster with 1.18, I point that to my storage location so that antrea bootstrap script gets called. Not sure if you facing any problem with 1.19 cluster due to this.

Can you describe that workflow in detail?

reachjainrahul commented:
> > Another thing to note, this change includes bootstrap script for windows and it needs to be signed before it can be installed. When I bring up the cluster with 1.18, I point that to my storage location so that antrea bootstrap script gets called. Not sure if you facing any problem with 1.19 cluster due to this.
> 
> 
> 
> Can you describe that workflow in detail?

This is not specific to Antrea. I think any PowerShell script needs to be signed by Microsoft. For developers to test addition of ps script like windowsantreacnifunc.ps1, we use build provisioning script and provide a storage location. And in cluster.json, that location is provided. 

This workflow is specific to Aks engine.

jackfrancis commented:
We can't consider merging this until it works. As it stands, installing the antrea addon with a Windows node pool doesn't produce working Windows nodes.

reachjainrahul commented:
> We can't consider merging this until it works. As it stands, installing the antrea addon with a Windows node pool doesn't produce working Windows nodes.

@jsturtevant @jackfrancis whats the workflow to add PowerShell script like windowsantreacnifunc.ps1. 

May I know, how is provisioning script called in the test env ? And what happens when someone make changes to windows provisioning script. How does e2e tests picks up the change? 

Also Jack, I did verify cluster with 1.18 . And it works. Even I ran E2E tests on it.

jackfrancis commented:
@reachjainrahul I am testing 1.18 now

reachjainrahul commented:
> @reachjainrahul I am testing 1.18 now

This is what I did.
I first ran build-windows-provisioning-scripts.sh and copied the location. Updated cluster.json with the location, such as
"provisioningScriptsPackageURL": "https://XXXX.blob.core.windows.net/rjakscontainer/windows-provisioning-scripts-1600060694.zip"

I am running the test with 1.19 now.. Lets see

jackfrancis commented:
@reachjainrahul so you're saying that this cluster config can't be tested as-is?

https://github.com/Azure/aks-engine/pull/3597/files#diff-629aa0331a7eeabadc497a6ad58c16ac

(That would explain why things are not testing out as working.)

@jsturtevant @marosset are you aware of need of having a custom built `provisioningScriptsPackageURL` config for the Windows antrea solution?

Is that requirement documented @reachjainrahul ?

reachjainrahul commented:
> @reachjainrahul so you're saying that this cluster config can't be tested as-is?
> 
> https://github.com/Azure/aks-engine/pull/3597/files#diff-629aa0331a7eeabadc497a6ad58c16ac
> 
> (That would explain why things are not testing out as working.)
> 
> @jsturtevant @marosset are you aware of need of having a custom built `provisioningScriptsPackageURL` config for the Windows antrea solution?
> 
> Is that requirement documented @reachjainrahul ?

Windows Provisioning script is changed in this PR. We introduced new file windowsantreacnifunc.ps1 for installing OVS and Antrea.  As per this documentation, inorder to test the change, we need to add provisioningScriptsPackageURL config https://github.com/Azure/aks-engine/blob/master/docs/topics/windows-provisioning-scripts.md

I am not sure, how in CI (E2E test), this new change will be picked up. If the change is not picked, antrea wont be installed on worker node.

reachjainrahul commented:
@jackfrancis @jsturtevant @mboersma 

As part of this checkin, we are modifying master/parts/k8s/kuberneteswindowssetup.ps1 which is the bootstrap powershell script used to bring up windows node. Like install docker, CNI, OpenSSL and so on. Antrea installation happens via that script

As per this documentation from AKS Engine, any powershell script needs to be signed. So, even kuberneteswindowssetup.ps1 is signed by Microsoft and put at some predefined location. 
For developers to add new cni install or modify kuberneteswindowssetup.ps1, the steps are documented here. 
https://github.com/Azure/aks-engine/blob/master/docs/topics/windows-provisioning-scripts.md

Basically zip your files and put it under some storage location and modify cluster.json with  'provisioningScriptsPackageURL' and provide the location. This requirement is not from Antrea, but from AKS-Engine. 

It would be good to understand, if someone modifies this file kuberneteswindowssetup.ps1, whose responsibility is to sign the files @jsturtevant and upload it to storage. I believe it's out of band operation and done manually.  

The problem @jackfrancis is facing with E2E test is; this change of antrea installation is not picked up, as kuberneteswindowssetup.ps1  is still pointing to checked in version, rather than new version available with this change.

reachjainrahul commented:
> @jackfrancis @jsturtevant @mboersma 
> 
> 
> 
> As part of this checkin, we are modifying master/parts/k8s/kuberneteswindowssetup.ps1 which is the bootstrap powershell script used to bring up windows node. Like install docker, CNI, OpenSSL and so on. Antrea installation happens via that script
> 
> 
> 
> As per this documentation from AKS Engine, any powershell script needs to be signed. So, even kuberneteswindowssetup.ps1 is signed by Microsoft and put at some predefined location. 
> 
> For developers to add new cni install or modify kuberneteswindowssetup.ps1, the steps are documented here. 
> 
> https://github.com/Azure/aks-engine/blob/master/docs/topics/windows-provisioning-scripts.md
> 
> 
> 
> Basically zip your files and put it under some storage location and modify cluster.json with  'provisioningScriptsPackageURL' and provide the location. This requirement is not from Antrea, but from AKS-Engine. 
> 
> 
> 
> It would be good to understand, if someone modifies this file kuberneteswindowssetup.ps1, whose responsibility is to sign the files @jsturtevant and upload it to storage. I believe it's out of band operation and done manually.  
> 
> 
> 
> The problem @jackfrancis is facing with E2E test is; this change of antrea installation is not picked up, as kuberneteswindowssetup.ps1  is still pointing to checked in version, rather than new version available with this change. 
> 
> 
> 
> 
> 
> 

Any updates ??

jsturtevant commented:
@reachjainrahul sorry for the delay. Could you re-base the changes. And I can generate a package that I can give to @jackfrancis for the e2e. If we get a good signal for those then we can merge and I will kick off the out of band signing process.

reachjainrahul commented:
> @reachjainrahul sorry for the delay. Could you re-base the changes. And I can generate a package that I can give to @jackfrancis for the e2e. If we get a good signal for those then we can merge and I will kick off the out of band signing process.

I rebased and ran E2E tests. All passed.

jsturtevant commented:
it looks like the ci provisioning script test was a not related to these changes. Re-triggered.

jsturtevant commented:
@jackfrancis you can use https://pssigned11493.blob.core.windows.net/ps-signed/windows-provisioning-scripts-1604682769.zip  

with `-e WINDOWS_PROVISIONING_SCRIPTS_URL=${SCRIPT_PACKAGE_URL} make test-kubernetes`

marosset commented:
@jackfrancis @jsturtevant If we need another way to test these changes we can expose the cluster definition file as a Azure DevOps pipeline variable and trigger runs that test changes to the provisioning scripts along with different cluster shapes.

jackfrancis commented:
@reachjainrahul I'm not seeing anything scheduled on the Windows node in the test config:

```
NAMESPACE     NAME                                                READY   STATUS      RESTARTS   AGE     IP             NODE                               NOMINATED NODE   READINESS GATES
 default       pod/busybox-agent-m7wns                             0/1     Completed   0          2m6s    10.244.1.5     k8s-agentlin-59447616-vmss000000   <none>           <none>
 default       pod/busybox-master-4pqgv                            0/1     Completed   0          2m37s   10.244.0.3     k8s-master-59447616-0              <none>           <none>
 default       pod/windows-schedule-test                           0/1     Completed   0          84s     10.244.2.2     5944k8s01000000                    <none>           <none>
 kube-system   pod/antrea-agent-6b2l2                              2/2     Running     0          7m33s   10.240.0.5     k8s-agentlin-59447616-vmss000000   <none>           <none>
 kube-system   pod/antrea-agent-9hcb5                              2/2     Running     0          7m33s   10.240.255.5   k8s-master-59447616-0              <none>           <none>
 kube-system   pod/antrea-controller-8656799c67-g46tz              1/1     Running     0          7m33s   10.240.0.5     k8s-agentlin-59447616-vmss000000   <none>           <none>
 kube-system   pod/blobfuse-flexvol-installer-wx2kl                1/1     Running     0          6m51s   10.244.1.3     k8s-agentlin-59447616-vmss000000   <none>           <none>
 kube-system   pod/coredns-56db558fb9-hqc76                        1/1     Running     0          7m38s   10.244.1.2     k8s-agentlin-59447616-vmss000000   <none>           <none>
 kube-system   pod/coredns-autoscaler-5c7db64899-2rxzh             1/1     Running     0          7m37s   10.244.0.2     k8s-master-59447616-0              <none>           <none>
 kube-system   pod/csi-secrets-store-22v87                         3/3     Running     0          6m51s   10.240.0.5     k8s-agentlin-59447616-vmss000000   <none>           <none>
 kube-system   pod/csi-secrets-store-provider-azure-jsjl7          1/1     Running     0          6m51s   10.240.0.5     k8s-agentlin-59447616-vmss000000   <none>           <none>
 kube-system   pod/kube-addon-manager-k8s-master-59447616-0        1/1     Running     0          7m43s   10.240.255.5   k8s-master-59447616-0              <none>           <none>
 kube-system   pod/kube-apiserver-k8s-master-59447616-0            1/1     Running     0          7m1s    10.240.255.5   k8s-master-59447616-0              <none>           <none>
 kube-system   pod/kube-controller-manager-k8s-master-59447616-0   1/1     Running     0          7m5s    10.240.255.5   k8s-master-59447616-0              <none>           <none>
 kube-system   pod/kube-proxy-jptdv                                1/1     Running     0          7m35s   10.240.255.5   k8s-master-59447616-0              <none>           <none>
 kube-system   pod/kube-proxy-sns2x                                1/1     Running     0          7m35s   10.240.0.5     k8s-agentlin-59447616-vmss000000   <none>           <none>
 kube-system   pod/kube-scheduler-k8s-master-59447616-0            1/1     Running     0          7m14s   10.240.255.5   k8s-master-59447616-0              <none>           <none>
 kube-system   pod/metrics-server-b6dbc49fc-nsjp7                  1/1     Running     0          7m35s   10.244.1.4     k8s-agentlin-59447616-vmss000000   <none>           <none>
```

Is that expected?

reachjainrahul commented:
> @reachjainrahul I'm not seeing anything scheduled on the Windows node in the test config:
> 
> ```
> NAMESPACE     NAME                                                READY   STATUS      RESTARTS   AGE     IP             NODE                               NOMINATED NODE   READINESS GATES
>  default       pod/busybox-agent-m7wns                             0/1     Completed   0          2m6s    10.244.1.5     k8s-agentlin-59447616-vmss000000   <none>           <none>
>  default       pod/busybox-master-4pqgv                            0/1     Completed   0          2m37s   10.244.0.3     k8s-master-59447616-0              <none>           <none>
>  default       pod/windows-schedule-test                           0/1     Completed   0          84s     10.244.2.2     5944k8s01000000                    <none>           <none>
>  kube-system   pod/antrea-agent-6b2l2                              2/2     Running     0          7m33s   10.240.0.5     k8s-agentlin-59447616-vmss000000   <none>           <none>
>  kube-system   pod/antrea-agent-9hcb5                              2/2     Running     0          7m33s   10.240.255.5   k8s-master-59447616-0              <none>           <none>
>  kube-system   pod/antrea-controller-8656799c67-g46tz              1/1     Running     0          7m33s   10.240.0.5     k8s-agentlin-59447616-vmss000000   <none>           <none>
>  kube-system   pod/blobfuse-flexvol-installer-wx2kl                1/1     Running     0          6m51s   10.244.1.3     k8s-agentlin-59447616-vmss000000   <none>           <none>
>  kube-system   pod/coredns-56db558fb9-hqc76                        1/1     Running     0          7m38s   10.244.1.2     k8s-agentlin-59447616-vmss000000   <none>           <none>
>  kube-system   pod/coredns-autoscaler-5c7db64899-2rxzh             1/1     Running     0          7m37s   10.244.0.2     k8s-master-59447616-0              <none>           <none>
>  kube-system   pod/csi-secrets-store-22v87                         3/3     Running     0          6m51s   10.240.0.5     k8s-agentlin-59447616-vmss000000   <none>           <none>
>  kube-system   pod/csi-secrets-store-provider-azure-jsjl7          1/1     Running     0          6m51s   10.240.0.5     k8s-agentlin-59447616-vmss000000   <none>           <none>
>  kube-system   pod/kube-addon-manager-k8s-master-59447616-0        1/1     Running     0          7m43s   10.240.255.5   k8s-master-59447616-0              <none>           <none>
>  kube-system   pod/kube-apiserver-k8s-master-59447616-0            1/1     Running     0          7m1s    10.240.255.5   k8s-master-59447616-0              <none>           <none>
>  kube-system   pod/kube-controller-manager-k8s-master-59447616-0   1/1     Running     0          7m5s    10.240.255.5   k8s-master-59447616-0              <none>           <none>
>  kube-system   pod/kube-proxy-jptdv                                1/1     Running     0          7m35s   10.240.255.5   k8s-master-59447616-0              <none>           <none>
>  kube-system   pod/kube-proxy-sns2x                                1/1     Running     0          7m35s   10.240.0.5     k8s-agentlin-59447616-vmss000000   <none>           <none>
>  kube-system   pod/kube-scheduler-k8s-master-59447616-0            1/1     Running     0          7m14s   10.240.255.5   k8s-master-59447616-0              <none>           <none>
>  kube-system   pod/metrics-server-b6dbc49fc-nsjp7                  1/1     Running     0          7m35s   10.244.1.4     k8s-agentlin-59447616-vmss000000   <none>           <none>
> ```
> 
> Is that expected?

Antrea for windows doesnt run as containers (POD). It runs as native windows process.
